src/server_manager/web_app/app.ts

```diff
@@ -653,23 +653,23 @@ export class App {
   }
 
   public async showServer(server: server.Server): Promise<void> {
-    const serverId = localServerId(server);
     this.selectedServer = server;
-    this.appRoot.selectedServerId = serverId;
-    localStorage.setItem(LAST_DISPLAYED_SERVER_STORAGE_KEY, serverId);
-
-    if (isManagedServer(server)) {
-      if (!(server as server.ManagedServer).isInstallCompleted()) {
-        this.appRoot.showProgress(this.makeDisplayName(server), true);
-        return;
-      }
-    }
+    this.appRoot.selectedServerId = localServerId(server);
+    this.setLastDisplayedServer(server);
     await this.updateServerView(server);
     this.appRoot.showServerView();
   }
 
+  setLastDisplayedServer(server: server.Server) {
+    localStorage.setItem(LAST_DISPLAYED_SERVER_STORAGE_KEY, localServerId(server));
+  }
+
   private async updateServerView(server: server.Server): Promise<void> {
-    if (await server.isHealthy()) {
+    if (isManagedServer(server) && !(server as server.ManagedServer).isInstallCompleted()) {
+      await sleep(1000);  // TODO: Wait for view to be initialized.
```

Comment:
This is odd to me. We shouldn't need this kind of sleep. What exactly are you waiting for? Could you clarify why you need this?

Reply:
Removed.